### ISSUE TITLE
Speed up PWM

### ISSUE THREAD

**Issue by jake653** [s-light/slight_CircuitPython_TLC5957#1]
I was interested in a Persistence of Vision project, can someone verify that the Poker Mode which adjusts the PWM data size does not? change the PWM output timing (I fear it is always taking 65536 Gclock cycles)

**Comment by s-light** [comment]
hi @jake653, 

Tldr: if i understand it correctly it displays only as much bits as you send.

its a while that i have experimented with this..
and if i remember correctly the poker mode is not implemented in the library yet at all...
or in the mean time you can use the [arduino lib](https://github.com/s-light/slight_TLC5957) with this i think it is fast enough without poker mode.
feel free to create a pull request ;-)

i think CircuitPython is a bit slow for POV - as you really need mega fast updates
and the chip has the issue that it needs the latch puls timed to the clock of the Data.
and currently i have implemented this part as bit-banging - so it is relative slow..

## Details:

> [9.3.6 Poker Mode](http://www.ti.com/lit/ds/symlink/tlc5957.pdf#page=15&zoom=auto,-16,375)
> Poker Mode provides the TLC5957 with a flexible PWMbit, from9 bit to 16 bit. Therefore, data length can be reduced. In high multiplexing applications, Poker Mode can significantly increase visual refresh rate

in the AppNote there are more information: [3.4.3PokerModefor GS Data](http://www.ti.com/lit/ug/slvuaf0/slvuaf0.pdf#page=17&zoom=auto,-117,236)

and i also did a quick [search in the TI forum](https://e2e.ti.com/search?q=TLC5957%20Poker%20Mode&category=forum):
[TLC5957: How to input GSCLK](https://e2e.ti.com/support/power-management/f/196/p/791376/2948145?tisearch=e2e-sitesearch&keymatch=TLC5957%20Poker%20Mode#pi320995filter=all&pi320995scroll=false)

> [15/4/2019 10:52 Sean Lin1](https://e2e.ti.com/support/power-management/f/196/p/791376/2932592#2932592)
>Hi Kenneth,
>
>Q1. As GS Counter is 16 bit, if we enable "Auto Display Repeat" & use continuous GSCLK, below two options which one is correct?
>1. The OUTn PWM On/Off signal will auto repeat according to PWM bit count
>ex: 16 bits PWM will repeat every 2^16 GSCLKs
>12 bits PWM will repeat every 2^12 GSCLKs
>10 bits PWM will repeat every 2^ 10 GSCLKs
>...
>2. No matter the PWM bit count, it will all auto repeat PWM On/Off signal every 2^16 GSCLKs
>=> if this is the case, when using 10 bits PWM for example, will OUTn turn off between GSCLK >1024 th ~ 65534 th?
>
>Q2. As GS Counter is 16 bit, if we enable "Auto Display Repeat" & use continuous GSCLK, is the GS data only updated & applied every 65535 GSCLK
>NO MATTER the PWM bits count?
>(if so, i think it's the reason you say how it effect the refresh rate)
>
>Q3. If we enable "Auto Display Repeat" & use continuous GSCLK, Do u suggest we use as higher frequency as better?
>will the frequency we input GSCLK effect the LED drive result?

> [16/4/2019 9:35 Kenneth Du](https://e2e.ti.com/support/power-management/f/196/p/791376/2934343#2934343)
> Q1. Option 1 is correct. If you use 12 bits, it repeats every 2^12 GSCLKs.
**Q2. You can use poker mode to send N(9-16) bits data, and the total period is 2^N GSCLKs. Thus you can improve the refresh rate.**
Q3. The output PWM frequency = GSCLK Frequency/2^N * 128. So higher GSCLK results to higher OUTPUT PWM frequency.
>
>Thanks and best regards,
>
>Kenneth

## other things
my idea was also to use this driver for POV..
and the poker mode was one reason i choose it ;-)
i have not tested it in this configuration -
but have some boards laying around..
you can find all the information i collected over time regarding POV at my ['ortogere project'](https://s-light.github.io/ortogere/docs/ideas/POV/pov_leddriver.html)

i also have a [TLC5951](http://www.ti.com/product/TLC5951) laying on my table -
it could have some really nice options for POV...
but i did not tried anything with it yet..

**Comment by s-light** [comment]
additionally regarding PWM-Speed have a look at https://github.com/adafruit/circuitpython/pull/1363
